client/src/auth/Signup.js

```diff
@@ -0,0 +1,87 @@
+
+import React from "react";
+
+class Signup extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      username: "",
+      password: "",
+      password2: "",
+      department: "",
+      error: ""
+    };
+  }
+
+  changeHandler = e => {
+    return this.setState({ [e.target.name]: e.target.value });
+  };
+
+  registerUser = e => {
+    e.preventDefault();
+    const user = {
+      username: this.state.username,
+      password: this.state.password,
+      department: this.state.department
```

Comment:
One way to shorten pulling these things off of state with destructuring would be:
```javascript
const { username, password, department } = this.state;
const user = { username, password, department };
```